### PR TITLE
fix(ui) Fix collapsing headers in PanelTable

### PR DESCRIPTION
**Pull request by markstory**
My previous change to PanelTableHeader caused empty header cells to collapse. This reverts that change and add storybook coverage to that scenario and the SimpleTableChart component.

### CODE REVIEW

File: docs-ui/components/panels.stories.js [hunk 1]
```diff
@@ -71,6 +71,17 @@ export const _PanelTable = withInfo({
       <div>Panel Item</div>
       <div>Panel Item</div>
       <div>Panel Item</div>
+      <div>Panel Item</div>
+      <div>Panel Item</div>
+    </PanelTable>
+
+    <PanelTable
+      // eslint-disable-next-line react/jsx-key
```
Comment (k-fish) [inline docs-ui/components/panels.stories.js:79]: lint disable... well it's in a story code so I'm chill with it.

File: src/sentry/static/sentry/app/components/charts/simpleTableChart.tsx [hunk 2]
```diff
@@ -68,10 +70,10 @@ const StyledPanelTable = styled(PanelTable)`
   border-right: 0;
   border-bottom: 0;
 
-  /* align height with charts */
-  height: 190px;
-  overflow: scroll;
   margin: ${space(1)} 0 0;
+  ${/* sc-selector */ PanelTableHeader} {
```
Comment (k-fish) [inline src/sentry/static/sentry/app/components/charts/simpleTableChart.tsx:74]: Is this intentional? Is this some sort of macro?

**Review by k-fish — APPROVED** [review]

**Review by priscilawebdev — APPROVED** [review]
lgtm

**Comment by priscilawebdev** [comment]
Thanks for fixing that 🙏

File: docs-ui/components/panels.stories.js (hunk 1 above)
Comment (priscilawebdev) [inline docs-ui/components/panels.stories.js:79]: I think in case of string you don't need to add a key... so its fine to remove this lint disable

Comment (priscilawebdev) [reply docs-ui/components/panels.stories.js:79]: image

File: src/sentry/static/sentry/app/components/charts/simpleTableChart.tsx (hunk 2 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/charts/simpleTableChart.tsx:74]: It is a [hint to stylelint/eslint to not yell about selectors](https://styled-components.com/docs/tooling#interpolation-tagging) with objects in them.

**Pull request merged**